common/src/main/java/dev/isxander/yacl3/gui/controllers/dropdown/ItemControllerElement.java

```diff
@@ -0,0 +1,70 @@
+package dev.isxander.yacl3.gui.controllers.dropdown;
+
+import dev.isxander.yacl3.api.utils.Dimension;
+import dev.isxander.yacl3.gui.YACLScreen;
+import dev.isxander.yacl3.gui.utils.ItemRegistryHelper;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+
+import java.util.List;
+
+public class ItemControllerElement extends AbstractDropdownControllerElement<Item, ResourceLocation> {
+	private final ItemController itemController;
+
+	public ItemControllerElement(ItemController control, YACLScreen screen, Dimension<Integer> dim) {
+		super(control, screen, dim);
+		this.itemController = control;
+	}
+
+	@Override
+	protected void drawValueText(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+		var oldDimension = getDimension();
+		setDimension(getDimension().withWidth(getDimension().width() - getDecorationPadding()));
+		super.drawValueText(graphics, mouseX, mouseY, delta);
+		setDimension(oldDimension);
+		if (ItemRegistryHelper.isRegisteredItem(inputField)) {
+			graphics.renderFakeItem(new ItemStack(ItemRegistryHelper.getItemFromName(inputField)), getDimension().xLimit() - getXPadding() - getDecorationPadding() + 2, getDimension().y() + 2);
```

Comment:
Again here, could the `ItemStack` be cached from input change? Accessing the registry every frame probably isn't the best.